Stratis.Bitcoin/AsyncLoopFactory.cs

```diff
@@ -17,18 +18,21 @@ public interface IAsyncLoopFactory
 
     public class AsyncLoopFactory : IAsyncLoopFactory
     {
-        public AsyncLoopFactory()
+        private readonly ILogger logger;
+
+        public AsyncLoopFactory(ILoggerFactory loggerFactory)
         {
+            this.logger = loggerFactory.CreateLogger("Stratis.Bitcoin.FullNode");
         }
 
         public IAsyncLoop Create(string name, Func<CancellationToken, Task> loop)
         {
-            return new AsyncLoop(name, loop);
+            return new AsyncLoop(name, this.logger, loop);
```

Comment:
I'm not a big fan of having the logger passed into all these methods. This purely functional method has some logging to it.  maybe the logging should be done in the caller method?